ecr-login/api/factory.go

```diff
@@ -109,3 +120,12 @@ func getServiceEndpoint(service, region string) (string, error) {
 	})
 	return endpoint.URL, err
 }
+
+func loadSharedConfigState() session.SharedConfigState {
+	awsSdkLoadConfig := os.Getenv("AWS_SDK_LOAD_CONFIG")
+	enableSharedConfig, _ := strconv.ParseBool(awsSdkLoadConfig)
+	if !enableSharedConfig && awsSdkLoadConfig != "" {
+		return session.SharedConfigDisable
+	}
+	return session.SharedConfigEnable
```

Comment:
I'd recommend a change like this:

```go
loadConfig := os.Getenv("AWS_SDK_LOAD_CONFIG")
if loadConfig == "" {
	return session.SharedConfigEnable
}
if enable, err := strconv.ParseBool(loadConfig); err == nil && !enable {
	return session.SharedConfigDisable
}
return session.SharedConfig
```

This gives you:

* Slightly shorter variable names (generally good in Go, especially in short functions)
* Removing ambiguity around "awsSdk" vs "awsSDK" (should be [capitalized in Go](https://github.com/golang/go/wiki/CodeReviewComments#initialisms))
* Errors not ignored
* More explicit that nothing defined means enabled and only explicitly setting to false is disabled